build/scripts/cibuild.ps1

```diff
@@ -0,0 +1,144 @@
+[CmdletBinding(PositionalBinding=$false)]
+param (
+    [switch]$test64 = $false,
+    [switch]$testDeterminism = $false,
+    [switch]$testBuildCorrectness = $false,
+    [switch]$testPerfCorrectness = $false,
+    [switch]$testPerfRun = $false,
+    [switch]$testVsi = $false,
+    [switch]$skipTest = $false,
+    [switch]$skipRestore = $false,
+    [switch]$skipCommitPrinting = $false,
+    [switch]$release = $false
+)
+
+Set-StrictMode -version 2.0
+$ErrorActionPreference = "Stop"
+
+function Print-Usage() {
+    Write-Host "Usage: cibuild.cmd [-debug^|-release] [-test32^|-test64] [-restore]"
+    Write-Host "  -debug   Perform debug build.  This is the default."
+    Write-Host "  -release Perform release build."
+    Write-Host "  -test32  Run unit tests in the 32-bit runner.  This is the default."
+    Write-Host "  -test64  Run units tests in the 64-bit runner."
+}
+
+function Run-MSBuild() {
+    # Because override the C#/VB toolset to build against our LKG package, it is important
+    # that we do not reuse MSBuild nodes from other jobs/builds on the machine. Otherwise,
+    # we'll run into issues such as https://github.com/dotnet/roslyn/issues/6211.
+    # MSBuildAdditionalCommandLineArgs=
+    & $msbuild /nologo /m /nodeReuse:false /consoleloggerparameters:Verbosity=minimal /filelogger /fileloggerparameters:Verbosity=normal @args
+    if (-not $?) {
+        throw "Build failed"
+    }
+}
+
+# Kill any instances VBCSCompiler.exe to release locked files, ignoring stderr if process is not open
+# This prevents future CI runs from failing while trying to delete those files.
+# Kill any instances of msbuild.exe to ensure that we never reuse nodes (e.g. if a non-roslyn CI run
+# left some floating around).
+function Terminate-BuildProcesses() {
+    gps msbuild -ErrorAction SilentlyContinue | kill 
```

Comment:
It's the satellite network that tells you where you are. Why it's related to Roslyn is less clear here... :smile: